docs: update release process (#2944)
electron-userland/electron-forge: CONTRIBUTING.md

### Release process

> - If you aren't sure if a release should happen, open an issue

Reviewer: question: Is this still valid? I would assume this is for maintainers and I don't think we'd use the issue tracker to discuss potential releases.

Author: I agree- questions would not get fielded on this platform

Reviewer 2: Yeah, I think I would just remove this bullet; it's not particularly helpful.

- Make sure the tests pass
- `$ ./tools/bump.ts $NEW_VERSION`
  - This will commit the changes automatically. Run `git log` to confirm that the changes have been
    committed.

>   - `$NEW_VERSION` i.e. `6.0.0-beta.67` (no `v` prefix)

Reviewer: ```suggestion
  - `$NEW_VERSION` should be an un-prefixed [semantic version](https://semver.org/) number (e.g. `6.0.0-beta.67)
```

- `$ node tools/publish.js`
- Push to default branch

> - Do a github release

Reviewer: ```suggestion
- Create a new GitHub release
```

>   - this will create an appropriate tag.

Reviewer: note that `git push` won't automatically push the tag up. you need to `git push origin master` and `git push origin v1.0.0` separately.

Reviewer 2: Fiddle has a great CONTRIBUTING section for [running a new release](https://github.com/electron/fiddle/blob/main/CONTRIBUTING.md#release-process). Taking some verbiage from there:

```suggestion
- After running the command, you should have a commit updating the version field in the package.json file, as well as a git tag corresponding with the new version number.
- Push your commit upstream to main/default branch.
- Push your tag to the GitHub repository with git push origin <tag_name> .
```
Don't have to use exactly this phrasing, but maybe something like this?

  - Go to releases tab

>   - Draft new release and make a new tag

Reviewer: issue(blocking): `bump.ts` already creates the git tag locally. You should be able to push the tag up rather than creating a new tag manually.

The "Push to default branch" step should also involve pushing that tag up.

  - Target default branch

>   - Generate release notes by copying in CHANGELOG.md contents into release description, or use the
>     github button feature.

Reviewer: ```suggestion
  - Generate release notes by copying in CHANGELOG.md contents into release description or using
    GitHub's [automatically generated release notes](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes)
```
